coding_systems/base/tests/test_import_data_utils.py

```diff
@@ -323,3 +324,46 @@ def test_save_codelist_draft_updates_compatibility_multiple_releases(self):
             self.bnf_version_with_search.compatible_releases.first().database_alias
             == "bnf_import-data_20220901"
         )
+
+
+class TestUpdateCodelistVersionCompatibilityIsOrderInsensitive(
+    BaseCodingSystemDynamicDatabaseTestCase
+):
+    db_aliases = [
+        "bnf_import-data_20221101",
+    ]
+
+    @pytest.mark.usefixtures("_get_bnf_release", "_get_bnf_review_version_with_search")
+    def test_update_codelist_version_compatibility_is_order_insensitive(
+        self,
+    ):
+        # modify the cached hierarchy data such that order of list elements is reversed
+        modified_existing_cached_hierarchy = json.loads(
+            self.bnf_review_version_with_search.cached_hierarchy.data
+        )
+        modified_existing_cached_hierarchy["nodes"] = list(
+            reversed(modified_existing_cached_hierarchy["nodes"])
+        )
+        modified_existing_cached_hierarchy["child_map"] = {
+            k: list(reversed(v))
+            for k, v in modified_existing_cached_hierarchy["child_map"].items()
+        }
+        modified_existing_cached_hierarchy["parent_map"] = {
```

Comment:
Do you need to modify the nodes, child_map and parent_map in order for this test to fail? i.e is there a simpler way of achieving the same thing?

Reply:
I'd like at least nodes and one of the maps to ensure that the recursive function is properly casting everything it needs to to a set